### PR TITLE
Fix workflow warning

### PR DESCRIPTION
**Pull request by salexanian**
## Summary of Changes  

This PR fixes a number of documentation build warnings, including:

-broken internal references in the installation and workflow documentation,
-ambiguous MyST cross-references in the user guide,
-missing intersphinx configuration for scikit-learn references,
-several API/docstring cross-reference issues.

This branch is based on #1091 as discussed. Once #1091 is merged, I'll update this branch against the latest main so the PR contains only these changes.

Issue 1:

`/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/checkouts/main/docs/getting_started/install.md:27: WARNING: 'myst' cross-reference target not found: 'contributing' [myst.xref_missing]`

Improper relative path. Fixed by changing `please refer to [Contributing](contributing)` in line 27 of `install.md` to `please refer to [Contributing](../library/contributing)`.

Issue 2: `/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/envs/main/lib/python3.11/site-packages/chainladder/workflow/gridsearch.py:docstring of chainladder.workflow.gridsearch.GridSearch:39: WARNING: term not in glossary: 'n_jobs'`

This problem was due to two issues - `n_jobs` was missing from the `glossary.md` file, and the lack of the Sphinx 'glossary' directive in `glossary.md`.  I rectified this by: 1) adding the `glossary` directive to the `glossary.md` file and 2) amending the docstring for `n_jobs` by reducing it to the first descriptive sentence in the docstring, and then moving the more detailed remainder of the docstring to the `glossary.md` file in a new `n_jobs` entry.

Issue 3: `/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/envs/main/lib/python3.11/site-packages/chainladder/workflow/gridsearch.py:docstring of chainladder.workflow.gridsearch.Pipeline:4: WARNING: undefined label: 'pipeline_docs'`

This error was due to a broken reference. After finding the proper reference in the code, I modified line 164 of `gridsearch.py` to read `Read more in the :ref:User Guide <workflow:pipeline> `

Issue 4: `/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/envs/main/lib/python3.11/site-packages/chainladder/workflow/gridsearch.py:docstring of chainladder.workflow.gridsearch.Pipeline.fit:32: WARNING: undefined label: 'metadata_routing'`

This error was due to an inability for the docstring to import the `metadata_routing` label from the `sklearn` library due to a failure to load intersphinx properly.  I fixed this by adding new line 40 to `_config.yml` as follows `- sphinx.ext.intersphinx` and by adding the new lines 50-53 as follows to the same file:

```intersphinx_mapping:
      sklearn:
        - https://scikit-learn.org/stable/
        - null
 ```

Issue 5 and 6: `/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/envs/main/lib/python3.11/site-packages/chainladder/workflow/gridsearch.py:docstring of chainladder.workflow.gridsearch.Pipeline.fit_predict:34: WARNING: undefined label: 'metadata_routing'
/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/envs/main/lib/python3.11/site-packages/chainladder/workflow/gridsearch.py:docstring of chainladder.workflow.gridsearch.Pipeline.predict:30: WARNING: undefined label: 'metadata_routing'` 

Solved by the solution to Issue 4 above.

Issue 7: `/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/envs/main/lib/python3.11/site-packages/chainladder/workflow/voting.py:docstring of chainladder.workflow.voting.VotingChainladder:9: WARNING: undefined label: 'voting'`

Fixed by amending line 165 of `voting.py` to read: `Read more in the :ref:User Guide <workflow:votingchainladder>.` (the reference within the chevrons was wrong).

Issue 8: `/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/checkouts/main/docs/user_guide/development.ipynb:540023: WARNING: 'myst' cross-reference target not found: 'modules/generated/chainladder.MunichAdjustment' [myst.xref_missing]`

Fixed by correcting path from `modules/generated/chainladder.MunichAdjustment` to `../library/generated/chainladder.MunichAdjustment` in `development.ipynb`. Also fixed reference to BerquistSherman to add a link like for Munich (housekeeping change).

Issues 9-11: `/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/checkouts/main/docs/user_guide/development.ipynb:540026: WARNING: 'myst' cross-reference target not found: 'modules/generated/chainladder.MunichAdjustment' [myst.xref_missing]
/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/checkouts/main/docs/user_guide/development.ipynb:540030: WARNING: 'myst' cross-reference target not found: 'modules/generated/chainladder.MunichAdjustment' [myst.xref_missing]
/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/checkouts/main/docs/user_guide/development.ipynb:620011: WARNING: 'myst' cross-reference target not found: 'modules/generated/chainladder.MunichAdjustment' [myst.xref_missing]`

Solved by solution to issue 8 above.

Issues 12 and 13: `/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/checkouts/main/docs/user_guide/index.md:12: WARNING: more than one target found for 'myst' cross-reference triangle: could be :std:ref:`Triangles` or :std:doc:`Triangles` [myst.xref_ambiguous]
/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/checkouts/main/docs/user_guide/index.md:22: WARNING: more than one target found for 'myst' cross-reference triangle: could be :std:ref:`Triangle` or :std:doc:`Triangle` [myst.xref_ambiguous]`

Changed `[Triangle](triangle)` to `{doc}`Triangles <triangle>` for the first reference, and changed `**Classes**: **[Triangle](triangle)**` to `**Classes**: **{py:class}`chainladder.Triangle`**`. 

Issue 14: `/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/checkouts/main/docs/user_guide/index.md:29: WARNING: more than one target found for 'myst' cross-reference development: could be :std:doc:`Development` or :py:mod:`chainladder.development` [myst.xref_ambiguous]`

Similar change to issues 12 and 13 above.

Issues 15-19: `/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/checkouts/main/docs/user_guide/index.md:45: WARNING: more than one target found for 'myst' cross-reference tails: could be :std:doc:`Tail Estimation` or :py:mod:`chainladder.tails` [myst.xref_ambiguous]
/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/checkouts/main/docs/user_guide/index.md:61: WARNING: more than one target found for 'myst' cross-reference methods: could be :std:doc:`IBNR Models` or :py:mod:`chainladder.methods` [myst.xref_ambiguous]
/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/checkouts/main/docs/user_guide/index.md:79: WARNING: more than one target found for 'myst' cross-reference adjustments: could be :std:doc:`Adjustments` or :py:mod:`chainladder.adjustments` [myst.xref_ambiguous]
/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/checkouts/main/docs/user_guide/index.md:97: WARNING: more than one target found for 'myst' cross-reference workflow: could be :std:doc:`Workflow` or :py:mod:`chainladder.workflow` [myst.xref_ambiguous]`

Similar changes to issues 12 and 13 above.

Issue 20: I also fixed the following issue: `/home/docs/checkouts/readthedocs.org/user_builds/chainladder-python/envs/main/lib/python3.11/site-packages/chainladder/core/triangle.py:docstring of chainladder.core.triangle.Triangle:97: ERROR: Unknown target name: "ultimate".
looking for now-outdated files... none found`

by changing a reference to `ultimate_` to ` 'ultimate_' `

The warnings no longer show when I perform the build using `uv run jb build .`.

## Related GitHub Issue(s)  

#1091 and #1111

## Additional Context for Reviewers  

- [ ] I passed tests locally for both code (`uv run pytest`) and documentation changes (`uv run jb build docs --builder=custom --custom-builder=doctest`)

---

> [!NOTE]
> **Low Risk**
> Documentation and docstring-only changes; no runtime logic or test behavior is altered.
> 
> **Overview**
> This PR cleans up **documentation and docstring cross-references** so Jupyter Book / Sphinx builds stop emitting warnings about missing links, ambiguous MyST targets, and undefined glossary terms.
> 
> **Sphinx / intersphinx:** `sphinx.ext.intersphinx` and a **scikit-learn** mapping are added in `docs/_config.yml` and `docs/conf.py`, so sklearn docstring refs (e.g. `metadata_routing`) resolve. `numpydoc_show_class_members` is set explicitly in `conf.py`.
> 
> **Glossary:** `docs/library/glossary.md` wraps terms in `{glossary}` directives, fixes the package name typo (`chanladder` → `chainladder`), and adds an **`n_jobs`** entry. `GridSearch`’s `n_jobs` docstring now points to that glossary term instead of duplicating joblib details.
> 
> **Docstring `:ref` fixes:** `Pipeline` and `VotingChainladder` user-guide links use `workflow:pipeline` and `workflow:votingchainladder`. `Triangle` docstrings use proper ``ultimate_`` markup for Sphinx.
> 
> **Docs paths and MyST:** Install page Contributing link uses `../library/contributing`. User guide index cards use `{doc}`, `{py:class}`, and `{ref}` to avoid ambiguous `triangle` / `development` / etc. cross-refs. `development.ipynb` API links point to `../library/generated/...`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 4f82b7d46906d0fe1fe22ad38a3b2e22c2220c73. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by henrydingliu** [comment]
#1091 is merged to main

**Comment by henrydingliu** [comment]
@salexanian i just realized that the only place to test these fixes on our repo is directly on RTD. so i cleared out experimental. could you please merge in the latest main, then redirect this PR to experimental?

**Comment by salexanian** [comment]
> @salexanian i just realized that the only place to test these fixes on our repo is directly on RTD. so i cleared out experimental. could you please merge in the latest main, then redirect this PR to experimental?

@henrydingliu , no problem. Per your suggestion, I've merged the latest `main` into this branch and retargeted the PR to `experimental` for RTD validation.

File: docs/conf.py [hunk 1]
```diff
@@ -1,10 +1,3 @@
-import sys
-from pathlib import Path
-_DOCS_EXT_DIR = Path(__file__).resolve().parent / '_ext'
-_p = str(_DOCS_EXT_DIR)
-if _p not in sys.path:
-    sys.path.insert(0, _p)
-
```
[the remark this answers is not in the document]
Comment (salexanian) [reply docs/conf.py:7]: Fixed. `prep_sphinx_conf.py` restored the required `_ext` path and the documentation build passes.

File: docs/user_guide/index.md [hunk 2]
```diff
@@ -34,14 +34,15 @@ Tooling to generate loss development patterns
 ```
 +++
 
-**Algorithms**: [Development](development:development), [ClarkLDF](development:clarkldf), …
+**Algorithms**: {py:class}`chainladder.Development`,
```
Comment (henrydingliu) [inline docs/user_guide/index.md]: this is now pointing to the API. the itnent is to point it to a dedication section in the user guide pages.

Comment (henrydingliu) [reply docs/user_guide/index.md]: @salexanian a friendly reminder in case you haven't seen comment in your notification. no rush if you are busy

Comment (salexanian) [reply docs/user_guide/index.md]: Thank you @henrydingliu - I will get on it this weekend.

Comment (salexanian) [reply docs/user_guide/index.md]: @henrydingliu - resolved.  I have changed the link to an explicit {ref} to the Development user guide section and verified the docs build succeeds locally (`pytest` and `uv run jb build docs`).

**Comment by henrydingliu** [comment]
@salexanian there are a few more links in the same file that was pointing to sections in the user guide but are now pointing to the API pages. could you please fix all of them?

**Comment by salexanian** [comment]
> @salexanian there are a few more links in the same file that was pointing to sections in the user guide but are now pointing to the API pages. could you please fix all of them?

No problem @henrydingliu , I will get on it ASAP.

**Review by henrydingliu — APPROVED** [review]
